test/components/kubernetes/baremetal_kubelet_labels_test.go

```diff
@@ -0,0 +1,75 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build baremetal
+// +build e2e
+
+package kubernetes //nolint:testpackage
+
+import (
+	"context"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkglabels "k8s.io/apimachinery/pkg/labels"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+)
+
+func TestBaremetalNodeHasNodeSpecificLabels(t *testing.T) {
+	// This map is copied from file ci/baremetal/baremetal-cluster.lokocfg.envsubst
+	// the field is `node_specific_labels`.
+	// The node name keys are costructed as follows: <cluster-name>-<controller/worker>-<index>
+	nodeNamesAndExpectedLabelsInCI := map[string]map[string]string{
+		"mercury-controller-0": {
+			"testkey": "testvalue",
+		},
+		"mercury-worker-0": {
+			"ingressnode": "yes",
+		},
+		"mercury-worker-1": {
+			"storagenode": "yes",
+		},
+	}
+
+	client := testutil.CreateKubeClient(t)
+
+	nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Errorf("could not list nodes: %v", err)
```

Comment:
You should use `Fatalf` here, otherwise `nodes` might be nil and code will panic when you dereference `nodes` for `Items` field.